docs/photutils/psf.rst

```diff
@@ -28,10 +28,379 @@ model-fitting analysis, regardless to exactly what kind of model is actually
+    along the process, it has the advantage of clarity so that one can
+    keep track and easily differentiate input/outputs.
+
+
+Basic Usage
+^^^^^^^^^^^
+
+The `~photutils.psf.DAOPhotPSFPhotometry` is the core class to implement the
+DAOPHOT algorithm to perform PSF photometry in crowded fields.
+
+It basically encapsulates the loop "FIND, GROUP, NSTAR, SUBTRACT" in one place
+so that one can easily perform PSF photometry just by setting up a
+`~photutils.psf.DAOPhotPSFPhotometry` object.
+
+This class was implemented in such a way that it can be used as a callable
+function. The basic idea is illustrated as follows:
+
+.. doctest-skip::
+
+    >>> # create a DAOPhotPSFPhotometry object
+    >>> from photutils.psf import DAOPhotPSFPhotometry 
+    >>> my_photometry = DAOPhotPSFPhotometry(finder=my_finder,
+    ...                                      group_maker=my_group,
+    ...                                      bkg_estimator=my_bkg,
+    ...                                      psf_model=my_psf_model,
+    ...                                      fitter=my_fitter, niters=1,
+    ...                                      fitshape=(7,7))
+    >>> # get photometry results
+    >>> photometry_results, residual_image = my_photometry(image=my_image)
+
+Where ``my_finder``, ``my_group``, and ``my_bkg`` may be any suitable callable
+function. This approach allows one to customize every part of the photometry
+process provided that their input/output are compatible with the input/ouput
+expected by `~photutils.psf.DAOPhotPSFPhotometry` . See the API documentation
+for details.
+
+Performing PSF Photometry
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's take a look in a simple example with simulated stars whose PSF is
+assumed to be Gaussian.
+
+First let's create an image with four overlapping stars:
+    
+.. plot::
+    :include-source:
+    
+    from astropy.table import Table 
+    from photutils.datasets import make_random_gaussians, make_noise_image, make_gaussian_sources
+
+    sigma_psf = 2.0
+    sources = Table()
+    sources['flux'] = [700, 800, 700, 800]
+    sources['x_mean'] = [12, 17, 12, 17]
+    sources['y_mean'] = [15, 15, 20, 20]
+    sources['x_stddev'] = sigma_psf*np.ones(4)
+    sources['y_stddev'] = sources['x_stddev']
+    sources['theta'] = [0, 0, 0, 0]
+    sources['id'] = [1, 2, 3, 4]
+    tshape = (32, 32)
+    image = (make_gaussian_sources(tshape, sources) +
+             make_noise_image(tshape, type='poisson', mean=6.,
+                              random_state=1) +
+             make_noise_image(tshape, type='gaussian', mean=0.,
+                              stddev=2., random_state=1))
+
+    from matplotlib import rcParams
+    import matplotlib.pyplot as plt
+    rcParams['image.cmap'] = 'viridis'
+    rcParams['image.aspect'] = 1  # to get images with square pixels
+    rcParams['image.interpolation'] = 'nearest'
+    rcParams['image.origin'] = 'lower'
+    rcParams['font.size'] = 16
+    
```

Comment:
@mirca Is there some reason you aren't using these settings as keywords, e.g.:
`plt.imshow(image, cmap='viridis', interpolation='nearest', origin='lower')`.